src/geo/projection/globe_tile_transform.js

```diff
@@ -65,9 +66,11 @@ export default class GlobeTileTransform {
         return latLngToECEF(latFromMercatorY(mercY), lngFromMercatorX(mercX), 1.0);
     }
 
-    upVectorScale(id: CanonicalTileID): number {
+    upVectorScale(id: CanonicalTileID, latitude: number, worldSize: number): ElevationScale {
+        const pixelsPerMeterAtLat = mercatorZfromAltitude(1, latitude) * worldSize;
         const pixelsPerMeterECEF = mercatorZfromAltitude(1, 0.0) * 2.0 * GLOBE_RADIUS * Math.PI;
-        return pixelsPerMeterECEF * globeECEFNormalizationScale(globeTileBounds(id));
+
+        return {metersToTile: pixelsPerMeterECEF * globeECEFNormalizationScale(globeTileBounds(id)), metersToLabelSpace: pixelsPerMeterAtLat};
```

Comment:
There might be a lot of redundant calculation of `globeECEFNormalizationScale(globeTileBounds(id))`, which should be the same for every symbols of a tile. Should it be part of the tile transform?

Reply:
Similarly `const pixelsPerMeterECEF = mercatorZfromAltitude(1, 0.0) * 2.0 * GLOBE_RADIUS * Math.PI;` could be defined as a constant.

Reply:
Thanks for the comments! I'll move `pixelsPerMeterECEF` to outside of the scope. We should do a separate refactoring pass to reduce these redundant calculations as you pointed out